docs/internals/generic/_index.md

```diff
@@ -0,0 +1,215 @@
+If you use the generic subtype, your module needs to define any and all service functionality and pass it through `DoCommand`.
+
+Whenever possible, it is best to use an [existing service API](/services/) instead of generic so that you do not have to replicate code.
+If you want to use most of an existing API but need just a few other functions, try using the `DoCommand` endpoint and extra parameters to add custom functionality to an [existing subtype](/services/), instead of using the generic service.
+
+{{% /alert %}}
+
+## Supported models
+
+Before creating a new generic service, check whether one of the following [modular resources](#modular-resources) supports your use case.
+
+{{< readfile "/static/include/create-your-own-mr.md" >}}
+
+### Built-in models
+
+For configuration information, click on the model name:
+
+<!-- prettier-ignore -->
+Model | Description
+----- | -----------
+[`fake`](fake/) | A model used for testing a generic service.
+
+### Modular resources
+
+{{<modular-resources api="rdk:service:generic" type="generic">}}
+
+## Control your machine with Viam's client SDK libraries
+
+To get started using Viam's SDKs to connect to and control your machine, go to your machine's page on [the Viam app](https://app.viam.com), navigate to the **Code sample** tab, select your preferred programming language, and copy the sample code generated.
+
+{{% snippet "show-secret.md" %}}
+
+When executed, this sample code will create a connection to your machine as a client.
+Then control your machine programmatically by getting your `generic` service from the machine with `FromRobot` and adding API method calls, as shown in the following examples.
+
+Be sure to import the generic service package for the SDK you are using:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```python
+from viam.services.generic import Generic
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go
+import (
+  "go.viam.com/rdk/services/generic"
+)
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+```cpp
+#include <viam/sdk/services/generic/generic.hpp>
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## API
+
+The generic service supports the following method:
+
+{{< readfile "/static/include/services/apis/generic.md" >}}
+
+### DoCommand
+
+Execute model-specific commands.
+If you are implementing your own generic service and add features that have no built-in API method, you can access them with `DoCommand`.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `command` [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): The command to execute.
+
+**Returns:**
+
+- [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): Result of the executed command.
+
+```python {class="line-numbers linkable-line-numbers"}
+my_generic = Generic.from_robot(robot=robot, name="my_generic_service")
+
+raw_dict = {
+  "command": "raw",
+  "raw_input": "home"
+}
+await my_generic.do_command(raw_dict)
+```
```

Comment:
Python code for using generic service appears identical to that for generic component. Please correct if I am wrong!

Reply:
Looks right to me!